### PR TITLE
Mc ci logon

### PR DESCRIPTION
**Pull request by FBRTMaka**
@oceanzus @DanielJMaher 

Supporting PR for the cilogon

### CODE REVIEW

**Comment by oceanzus** [comment]
@FBRTMaka Looks like rauth didn't make it into the requirements file.

**Comment by FBRTMaka** [comment]
@oceanzus good catch, updating now

File: ooiservices/app/config.yml [hunk 1]
```diff
@@ -55,7 +56,9 @@ COMMON: &common
     ALFRESCO_REPO_ID:   'c161bc66-4f7e-4a4f-b5f2-aac9fbf1d3cd'
     ALFRESCO_USERNAME:  'admin'
     ALFRESCO_PASSWORD:  'admin'
-
+    # cilogon
+    CILOGON_ID: 'myproxy:oa4mp,2012:/client_id/84e7951a567243a62ae8bc3e13c12c8'
```
Comment (oceanzus) [inline ooiservices/app/config.yml]: I think we should obfuscate these parameters (security).  Please make them generic 'xxx...'

File: ooiservices/app/main/oauth.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+
+class CILogonSignIn(OAuthSignIn):
+    def __init__(self):
+        super(CILogonSignIn, self).__init__('cilogon')
+        self.service = OAuth2Service(
+            name='cilogon',
+            client_id=self.consumer_id,
+            client_secret=self.consumer_secret,
+            authorize_url='https://test.cilogon.org/authorize',
```
Comment (oceanzus) [inline ooiservices/app/main/oauth.py]: Let's move these to the config file as well, knowing that they will change early next year when cilogon is formally released.

File: ooiservices/app/main/oauth.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                'redirect_uri': self.get_callback_url()}
+        print data
+
+        oauth_session = self.service.get_auth_session(
+            data=data,
+            decoder=self.custom_decoder
+        )
+
+        # return oauth_session.client_id
+        ci_logon_url = 'https://test.cilogon.org/oauth2/userinfo'
```
Comment (oceanzus) [inline ooiservices/app/main/oauth.py]: Config file as well

**Comment by FBRTMaka** [comment]
@oceanzus cool cool, would you mind making a PR to this branch?  I'm done for the night. Otherwise I'll do clean up tomorrow.

**Comment by oceanzus** [comment]
@FBRTMaka I'm toast as well.  Hit it in the morning.

**Comment by FBRTMaka** [comment]
@oceanzus I hear ya!  But I'm up now so I'll get it cleaned up.

File: ooiservices/app/main/oauth.py (hunk 2 above)
Comment (FBRTMaka) [inline ooiservices/app/main/oauth.py]: @oceanzus This is where we can use that nicer cilogon page that you sent via FD, right?

**Comment by oceanzus** [comment]
@DanielJMaher @FBRTMaka Looks good to me.  Ready to merge when the timing is right.

**Pull request merged**